domains/transfer/README.md

```diff
@@ -58,3 +58,50 @@ This is the command for generating the migration SQL script for the API project:
 ```shell
 dotnet ef migrations script --idempotent --project Transfer.API/API --output migrations/API.sql
 ```
+
+## API Versioning Strategy
+
+### Overview
+In the Transfer Agreements Domain, we implement a strategic approach to API versioning. This is particularly crucial for managing breaking changes to the API contract, ensuring backward compatibility, and facilitating smooth transitions between API versions.
+
+### Versioning Methodology
+Our versioning process is designed to handle breaking changes effectively.
+Say you want to create a new api version. The following steps are taken:
+
+1. **Duplication of Controllers and DTOs:**
+
+    - We start by duplicating each controller and its associated data transfer objects (DTOs).
+    - After duplication the folder structure should look like this, with transfer as example:
+   ```
+    transfer/
+    ├── v2023_01_01/
+    │   ├── Controllers/
+    │   └── Dto/
+    └── v2023_11_11/
+        ├── Controllers/
+        └── Dto/
+   ```
+    - Necessary modifications are then applied to these duplicates to incorporate the required changes.
+
+2. **Deprecation of Old Controllers:**
+    - Previous versions of the controllers are marked as deprecated by setting the `Deprecated = true` flag in the `ApiVersion` annotation. An example is shown below:
+      ```csharp
+      [Authorize]
+      [ApiController]
+      [ApiVersion("20230101", Deprecated = true)] # <--- Append flag to the annotation
+      [Route("api/claim-automation")]
+      ```
+
+### Streamlining the Versioning Process
+To ease the versioning process, we utilize the `create_new_api_version.sh` Bash script:
+
+- This script automates the creation of new controller and DTO folders, following the naming convention "vYYYY_MM_DD" (e.g., "v2023_01_01").
+- Note: Manual deprecation of old API controllers is still required after running this script.
+
+### Testing Strategy for New API Versions
+Our approach to testing new API versions involves:
```

Comment:
Does this mean we will have duplictated tests?

Reply:
No. But you will have duplicated test file names. 

If you try to make a breaking change to an endpoint, create a new version folder, and controllerTests.cs file for that api, where you made the breaking change. Then include a test that specifically only tests that new contract.